portal-web/docroot/html/common/themes/top_head.jsp

```diff
@@ -32,6 +32,10 @@ long cssLastModifiedTime = PortalWebResourcesUtil.getLastModified(PortalWebResou
 
 <link data-senna-track="temporary" href="<%= HtmlUtil.escapeAttribute(PortalUtil.getStaticResourceURL(request, themeDisplay.getCDNDynamicResourcesHost() + PortalWebResourcesUtil.getContextPath(PortalWebResourceConstants.RESOURCE_TYPE_CSS) + "/main.css", cssLastModifiedTime)) %>" id="liferayPortalCSS" rel="stylesheet" type="text/css" />
 
+<c:if test="<%= themeDisplay.isSignedIn() %>">
```

Comment:
Hey @pat270, we already have standard mechanisms to add resources to this `top_head.jsp`. The functionality is called `TopHeadDynamicInclude`, and you can see an example of use [in the lodash module](https://github.com/liferay/liferay-portal/blob/fcb58066ee248c9956f9aa1e18ec879c16b0ff1b/modules/apps/frontend-js/frontend-js-lodash-web/src/main/java/com/liferay/frontend/js/lodash/web/internal/servlet/taglib/LodashTopHeadDynamicInclude.java) for example.

Adding things to this JSP (unless there's no other option) is discouraged because it is too low level.

Reply:
I would create a TopHeadDynamicInclude in frontend-css-web and you can inject any css file you need based on any custom logic. I think that would be the best way to continue with this setup.

I've discussed this with @dsanz and @edalgrin in slack, and proposed to ask the DXP squad to do it if you don't know how to.

Reply:
I think this makes more sense in another module than mixing it with the current responsibilities of frontend-css-web (serving CSS everywhere). 

I would create a new module and use the `TopHeadDynamicInclude` to add the css conditionally, that way a clear separation of concerns.

Reply:
This sounds good. I'll look into it, be prepared for me to ask questions.

Reply:
@pat270 let me know if you need help at all with this dynamic include